Cannot run zed_rtabmap.launch

Hello,
I have a zed2 camera and could not run rtabmap example on my Jetson TX2. The rtabmap freezes and does not show anything.

**logs and files** 

```roslaunch zed_rtabmap_example zed_rtabmap.launch -v
... logging to /home/nvidia/.ros/log/d34fb552-7131-11ea-a696-00044bc73598/roslaunch-nvidia-desktop-32725.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://nvidia-desktop:44019/

SUMMARY
========

PARAMETERS
 * /rosdistro: melodic
 * /rosversion: 1.14.3
 * /zed/rtabmap/Grid/3D: True
 * /zed/rtabmap/Grid/CellSize: 0.05
 * /zed/rtabmap/Grid/ClusterRadius: 0.1
 * /zed/rtabmap/Grid/DepthDecimation: 1
 * /zed/rtabmap/Grid/DepthRoiRatios: [0.0, 0.0, 0.0, 0.0]
 * /zed/rtabmap/Grid/FlatObstacleDetected: True
 * /zed/rtabmap/Grid/FootprintHeight: 2.0
 * /zed/rtabmap/Grid/FootprintLength: 0.18
 * /zed/rtabmap/Grid/FootprintWidth: 0.18
 * /zed/rtabmap/Grid/FromDepth: True
 * /zed/rtabmap/Grid/GroundIsObstacle: False
 * /zed/rtabmap/Grid/MapFrameProjection: False
 * /zed/rtabmap/Grid/MaxGroundAngle: 30.0
 * /zed/rtabmap/Grid/MaxGroundHeight: -0.4
 * /zed/rtabmap/Grid/MaxObstacleHeight: 0.1
 * /zed/rtabmap/Grid/MinClusterSize: 3
 * /zed/rtabmap/Grid/MinGroundHeight: -0.5
 * /zed/rtabmap/Grid/NoiseFilteringMinNeighbors: 5
 * /zed/rtabmap/Grid/NoiseFilteringRadius: 0.1
 * /zed/rtabmap/Grid/NormalK: 20
 * /zed/rtabmap/Grid/NormalsSegmentation: False
 * /zed/rtabmap/Grid/PreVoxelFiltering: True
 * /zed/rtabmap/Grid/RangeMax: 3.0
 * /zed/rtabmap/Grid/RangeMin: 0.7
 * /zed/rtabmap/Grid/RayTracing: True
 * /zed/rtabmap/GridGlobal/Eroded: False
 * /zed/rtabmap/GridGlobal/FootprintRadius: 0.18
 * /zed/rtabmap/GridGlobal/FullUpdate: True
 * /zed/rtabmap/GridGlobal/MaxNodes: 0
 * /zed/rtabmap/GridGlobal/MinSize: 1.0
 * /zed/rtabmap/GridGlobal/OccupancyThr: 0.55
 * /zed/rtabmap/GridGlobal/ProbClampingMax: 0.971
 * /zed/rtabmap/GridGlobal/ProbClampingMin: 0.1192
 * /zed/rtabmap/GridGlobal/ProbHit: 0.7
 * /zed/rtabmap/GridGlobal/ProbMiss: 0.4
 * /zed/rtabmap/GridGlobal/UpdateError: 0.01
 * /zed/rtabmap/Mem/IncrementalMemory: true
 * /zed/rtabmap/Mem/InitWMWithAllNodes: false
 * /zed/rtabmap/approx_sync: True
 * /zed/rtabmap/config_path: ~/.ros/rtabmap.cfg
 * /zed/rtabmap/database_path: ~/.ros/rtabmap.db
 * /zed/rtabmap/frame_id: base_link
 * /zed/rtabmap/map_frame_id: map
 * /zed/rtabmap/odom_frame_id: odom
 * /zed/rtabmap/odom_sensor_sync: False
 * /zed/rtabmap/odom_tf_angular_variance: 0.001
 * /zed/rtabmap/odom_tf_linear_variance: 0.001
 * /zed/rtabmap/publish_tf: True
 * /zed/rtabmap/queue_size: 10
 * /zed/rtabmap/scan_normal_k: 0
 * /zed/rtabmap/subscribe_depth: True
 * /zed/rtabmap/subscribe_odom_info: False
 * /zed/rtabmap/subscribe_rgbd: False
 * /zed/rtabmap/subscribe_scan: False
 * /zed/rtabmap/subscribe_scan_cloud: False
 * /zed/rtabmap/subscribe_stereo: False
 * /zed/rtabmap/subscribe_user_data: False
 * /zed/rtabmap/tf_delay: 0.02
 * /zed/rtabmap/wait_for_transform_duration: 0.2
 * /zed/zed_description: <?xml version="1....
 * /zed/zed_node/auto_exposure_gain: True
 * /zed/zed_node/auto_whitebalance: True
 * /zed/zed_node/brightness: 4
 * /zed/zed_node/contrast: 4
 * /zed/zed_node/depth/depth_downsample_factor: 1.0
 * /zed/zed_node/depth/depth_stabilization: 1
 * /zed/zed_node/depth/max_depth: 5.0
 * /zed/zed_node/depth/min_depth: 0.2
 * /zed/zed_node/depth/openni_depth_mode: 0
 * /zed/zed_node/depth/quality: 3
 * /zed/zed_node/depth/sensing_mode: 0
 * /zed/zed_node/depth_confidence: 50
 * /zed/zed_node/depth_texture_conf: 100
 * /zed/zed_node/exposure: 100
 * /zed/zed_node/gain: 100
 * /zed/zed_node/gamma: 8
 * /zed/zed_node/general/base_frame: base_link
 * /zed/zed_node/general/camera_flip: False
 * /zed/zed_node/general/camera_model: zed2
 * /zed/zed_node/general/camera_name: zed
 * /zed/zed_node/general/gpu_id: -1
 * /zed/zed_node/general/grab_frame_rate: 30
 * /zed/zed_node/general/resolution: 2
 * /zed/zed_node/general/self_calib: True
 * /zed/zed_node/general/serial_number: 0
 * /zed/zed_node/general/svo_compression: 2
 * /zed/zed_node/general/verbose: False
 * /zed/zed_node/general/zed_id: 0
 * /zed/zed_node/hue: 0
 * /zed/zed_node/mapping/fused_pointcloud_freq: 1.0
 * /zed/zed_node/mapping/mapping_enabled: False
 * /zed/zed_node/mapping/max_mapping_range: -1
 * /zed/zed_node/mapping/resolution: 0.05
 * /zed/zed_node/object_detection/confidence_threshold: 50
 * /zed/zed_node/object_detection/object_tracking_enabled: False
 * /zed/zed_node/object_detection/od_enabled: False
 * /zed/zed_node/object_detection/people_detection: False
 * /zed/zed_node/object_detection/vehicle_detection: False
 * /zed/zed_node/point_cloud_freq: 15.0
 * /zed/zed_node/pos_tracking/area_memory: True
 * /zed/zed_node/pos_tracking/area_memory_db_path: 
 * /zed/zed_node/pos_tracking/fixed_z_value: 0.0
 * /zed/zed_node/pos_tracking/floor_alignment: False
 * /zed/zed_node/pos_tracking/imu_fusion: True
 * /zed/zed_node/pos_tracking/init_odom_with_first_valid_pose: True
 * /zed/zed_node/pos_tracking/initial_base_pose: [0.0, 0.0, 0.0, 0...
 * /zed/zed_node/pos_tracking/map_frame: map
 * /zed/zed_node/pos_tracking/odometry_frame: odom
 * /zed/zed_node/pos_tracking/path_max_count: -1
 * /zed/zed_node/pos_tracking/path_pub_rate: 2.0
 * /zed/zed_node/pos_tracking/publish_map_tf: True
 * /zed/zed_node/pos_tracking/publish_tf: True
 * /zed/zed_node/pos_tracking/two_d_mode: False
 * /zed/zed_node/pub_frame_rate: 15.0
 * /zed/zed_node/saturation: 4
 * /zed/zed_node/sensors/sensors_timestamp_sync: False
 * /zed/zed_node/sharpness: 4
 * /zed/zed_node/stream: 
 * /zed/zed_node/svo_file: 
 * /zed/zed_node/video/extrinsic_in_camera_frame: True
 * /zed/zed_node/video/img_downsample_factor: 1.0
 * /zed/zed_node/whitebalance_temperature: 42

NODES
  /zed/
    rtabmap (rtabmap_ros/rtabmap)
    rtabmapviz (rtabmap_ros/rtabmapviz)
    rviz (rviz/rviz)
    zed_node (zed_wrapper/zed_wrapper_node)
    zed_state_publisher (robot_state_publisher/robot_state_publisher)

auto-starting new master
process[master]: started with pid [32743]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to d34fb552-7131-11ea-a696-00044bc73598
process[rosout-1]: started with pid [32756]
started core service [/rosout]
process[zed/zed_state_publisher-2]: started with pid [32764]
process[zed/zed_node-3]: started with pid [32765]
process[zed/rtabmapviz-4]: started with pid [304]
[ INFO] [1585426946.272446131]: Initializing nodelet with 6 worker threads.
process[zed/rtabmap-5]: started with pid [306]
process[zed/rviz-6]: started with pid [320]
[ INFO] [1585426946.426357238]: ********** Starting nodelet '/zed/zed_node' **********
[ INFO] [1585426946.426549172]: SDK version : 3.1.0
[ INFO] [1585426946.426636115]: *** PARAMETERS ***
[ INFO] [1585426946.427668840]:  * Camera Name			-> zed
[ INFO] [1585426946.428679742]:  * Camera Resolution		-> HD720
[ INFO] [1585426946.429616244]:  * Camera Grab Framerate	-> 30
[ INFO] [1585426946.430686601]:  * Gpu ID			-> -1
[ INFO] [1585426946.438856340]:  * Camera ID			-> -1
[ INFO] [1585426946.440303141]:  * Verbose			-> DISABLED
[ INFO] [1585426946.443690754]:  * Camera Flip			-> DISABLED
[ INFO] [1585426946.446549860]:  * Self calibration		-> ENABLED
[ INFO] [1585426946.449329032]:  * Camera Model by param	-> zed2
[ INFO] [1585426946.450694233]:  * Image resample factor	-> 1
[ INFO] [1585426946.452348616]:  * Extrinsic param. frame	-> X RIGHT - Y DOWN - Z FWD
[ INFO] [1585426946.453967383]:  * Depth quality		-> ULTRA
[ INFO] [1585426946.455581639]:  * Depth Sensing mode		-> STANDARD
[ INFO] [1585426946.459921178]:  * OpenNI mode			-> DISABLED
[ INFO] [1585426946.461392362]:  * Depth Stabilization		-> ENABLED
[ INFO] [1585426946.462448735]:  * Minimum depth		-> 0.2 m
[ INFO] [1585426946.463553300]:  * Maximum depth		-> 5 m
[ INFO] [1585426946.464913030]:  * Depth resample factor	-> 1
[ INFO] [1585426946.466586420]:  * Path rate			-> 2 Hz
[ INFO] [1585426946.468055653]:  * Path history size		-> 1
[ INFO] [1585426946.470683754]:  * Odometry DB path		-> 
[ INFO] [1585426946.475689654]:  * Spatial Memory		-> ENABLED
[ INFO] [1585426946.477368965]:  * IMU Fusion			-> ENABLED
[ INFO] [1585426946.479122738]:  * Floor alignment		-> DISABLED
[ INFO] [1585426946.480738338]:  * Init Odometry with first valid pose data -> ENABLED
[ INFO] [1585426946.482512079]:  * Two D mode			-> DISABLED
[ INFO] [1585426946.485945388]:  * Mapping			-> DISABLED
[ INFO] [1585426946.487921751]:  * Object Detection		-> DISABLED
[ INFO] [1585426946.488981324]:  * Sensors timestamp sync	-> DISABLED
[ INFO] [1585426946.490289823]:  * Sensors data freq		-> 400 Hz
[ INFO] [1585426946.492159051]:  * SVO input file: 		-> 
[ INFO] [1585426946.493149057]:  * SVO REC compression		-> H265 (HEVC)
[ INFO] [1585426946.501393099]:  * map_frame			-> map
[ INFO] [1585426946.501486890]:  * odometry_frame		-> odom
[ INFO] [1585426946.501543306]:  * base_frame			-> base_link
[ INFO] [1585426946.501587273]:  * camera_frame			-> zed_camera_center
[ INFO] [1585426946.501630953]:  * imu_link			-> zed_imu_link
[ INFO] [1585426946.501678824]:  * left_camera_frame		-> zed_left_camera_frame
[ INFO] [1585426946.501720008]:  * left_camera_optical_frame	-> zed_left_camera_optical_frame
[ INFO] [1585426946.501755880]:  * right_camera_frame		-> zed_right_camera_frame
[ INFO] [1585426946.501841735]:  * right_camera_optical_frame	-> zed_right_camera_optical_frame
[ INFO] [1585426946.501881670]:  * depth_frame			-> zed_left_camera_frame
[ INFO] [1585426946.501925510]:  * depth_optical_frame		-> zed_left_camera_optical_frame
[ INFO] [1585426946.501960805]:  * disparity_frame		-> zed_left_camera_frame
[ INFO] [1585426946.501994789]:  * disparity_optical_frame	-> zed_left_camera_optical_frame
[ INFO] [1585426946.502039205]:  * confidence_frame		-> zed_left_camera_frame
[ INFO] [1585426946.502078116]:  * confidence_optical_frame	-> zed_left_camera_optical_frame
[ INFO] [1585426946.503930961]:  * Broadcast odometry TF	-> ENABLED
[ INFO] [1585426946.505803326]:  * Broadcast map pose TF	-> ENABLED
[ INFO] [1585426946.506935314]:  * [DYN] Depth confidence	-> 50
[ INFO] [1585426946.507927752]:  * [DYN] Depth texture conf.	-> 100
[ INFO] [1585426946.508882750]:  * [DYN] pub_frame_rate		-> 15 Hz
[ INFO] [1585426946.509857875]:  * [DYN] point_cloud_freq	-> 15 Hz
[ INFO] [1585426946.510816010]:  * [DYN] brightness		-> 4
[ INFO] [1585426946.511728192]:  * [DYN] contrast		-> 4
[ INFO] [1585426946.512679862]:  * [DYN] hue			-> 0
[ INFO] [1585426946.513733611]:  * [DYN] saturation		-> 4
[ INFO] [1585426946.514792736]:  * [DYN] sharpness		-> 4
[ INFO] [1585426946.515769238]:  * [DYN] gamma			-> 8
[ INFO] [1585426946.516711980]:  * [DYN] auto_exposure_gain	-> ENABLED
[ INFO] [1585426946.519322225]:  * [DYN] auto_whitebalance	-> ENABLED
[ INFO] [1585426946.539806973]:  * Camera coordinate system	-> Right HANDED Z UP and X FORWARD
[ INFO] [1585426946.540100218]:  *** Opening ZED 2...
[ INFO] [1585426946.797916806]: rviz version 1.13.7
[ INFO] [1585426946.798037637]: compiled against Qt version 5.9.5
[ INFO] [1585426946.798075205]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1585426946.809321808]: Forcing OpenGl version 0.
[ INFO] [1585426947.543908334]: Starting node...
[ INFO] [1585426947.789287358]: Starting node...
[ INFO] [1585426947.862852388]: Stereo is NOT SUPPORTED
[ INFO] [1585426947.863174944]: OpenGl version: 4.6 (GLSL 4.6).
[ INFO] [1585426948.000277107]: Initializing nodelet with 6 worker threads.
[ INFO] [1585426948.348939159]: ZED connection -> SUCCESS
[ INFO] [1585426948.888108805]: /zed/rtabmap(maps): map_filter_radius          = 0.000000
[ INFO] [1585426948.888239364]: /zed/rtabmap(maps): map_filter_angle           = 30.000000
[ INFO] [1585426948.888721407]: /zed/rtabmap(maps): map_cleanup                = true
[ INFO] [1585426948.888926397]: /zed/rtabmap(maps): map_always_update          = false
[ INFO] [1585426948.889126587]: /zed/rtabmap(maps): map_empty_ray_tracing      = true
[ INFO] [1585426948.889305945]: /zed/rtabmap(maps): cloud_output_voxelized     = true
[ INFO] [1585426948.889490583]: /zed/rtabmap(maps): cloud_subtract_filtering   = false
[ INFO] [1585426948.889676629]: /zed/rtabmap(maps): cloud_subtract_filtering_min_neighbors = 2
[ INFO] [1585426948.894533699]: /zed/rtabmap(maps): octomap_tree_depth         = 16
[ INFO] [1585426948.949716519]: rtabmap: frame_id      = base_link
[ INFO] [1585426948.949858374]: rtabmap: odom_frame_id = odom
[ INFO] [1585426948.949903621]: rtabmap: map_frame_id  = map
[ INFO] [1585426948.949940837]: rtabmap: use_action_for_goal  = false
[ INFO] [1585426948.949977412]: rtabmap: tf_delay      = 0.020000
[ INFO] [1585426948.950017028]: rtabmap: tf_tolerance  = 0.100000
[ INFO] [1585426948.950071811]: rtabmap: odom_sensor_sync   = false
[ERROR] [1585426948.970965099]: Header from subscriber did not have the required elements: md5sum, topic, callerid
[ INFO] [1585426948.991696116]: Loading parameters from /home/nvidia/.ros/rtabmap.cfg
[ INFO] [1585426949.137313617]: Setting RTAB-Map parameter "Grid/3D"="true"
[ INFO] [1585426949.143376914]: Setting RTAB-Map parameter "Grid/CellSize"="0.05"
[ INFO] [1585426949.151529022]: Setting RTAB-Map parameter "Grid/ClusterRadius"="0.1"
[ INFO] [1585426949.159254958]: Setting RTAB-Map parameter "Grid/DepthDecimation"="1"
[ INFO] [1585426949.183931598]: Setting RTAB-Map parameter "Grid/FlatObstacleDetected"="true"
[ INFO] [1585426949.193022672]: rtabmapviz: Using configuration from "/home/nvidia/.ros/rtabmapGUI.ini"
[ INFO] [1585426949.200239621]: Setting RTAB-Map parameter "Grid/FootprintHeight"="2"
[ INFO] [1585426949.205078739]: Setting RTAB-Map parameter "Grid/FootprintLength"="0.18"
[ INFO] [1585426949.212675685]: Setting RTAB-Map parameter "Grid/FootprintWidth"="0.18"
[ INFO] [1585426949.217233813]: Setting RTAB-Map parameter "Grid/FromDepth"="true"
[ INFO] [1585426949.220544563]: Setting RTAB-Map parameter "Grid/GroundIsObstacle"="false"
[ INFO] [1585426949.227604714]: Setting RTAB-Map parameter "Grid/MapFrameProjection"="false"
[ INFO] [1585426949.237177927]: Setting RTAB-Map parameter "Grid/MaxGroundAngle"="30"
[ INFO] [1585426949.246989409]: Setting RTAB-Map parameter "Grid/MaxGroundHeight"="-0.4"
[ INFO] [1585426949.252919396]: Setting RTAB-Map parameter "Grid/MaxObstacleHeight"="0.1"
[ INFO] [1585426949.262384866]: Setting RTAB-Map parameter "Grid/MinClusterSize"="3"
[ INFO] [1585426949.272835158]: Setting RTAB-Map parameter "Grid/MinGroundHeight"="-0.5"
[ INFO] [1585426949.282190421]: Setting RTAB-Map parameter "Grid/NoiseFilteringMinNeighbors"="5"
[ INFO] [1585426949.289135501]: Setting RTAB-Map parameter "Grid/NoiseFilteringRadius"="0.1"
[ INFO] [1585426949.297040155]: Setting RTAB-Map parameter "Grid/NormalK"="20"
[ INFO] [1585426949.301122737]: Setting RTAB-Map parameter "Grid/NormalsSegmentation"="false"
[ INFO] [1585426949.304339120]: Setting RTAB-Map parameter "Grid/PreVoxelFiltering"="true"
[ INFO] [1585426949.312175903]: Setting RTAB-Map parameter "Grid/RangeMax"="3"
[ INFO] [1585426949.321436575]: Setting RTAB-Map parameter "Grid/RangeMin"="0.7"
[ INFO] [1585426949.324734717]: Setting RTAB-Map parameter "Grid/RayTracing"="true"
[ INFO] [1585426949.340365371]: Setting RTAB-Map parameter "GridGlobal/Eroded"="false"
[ INFO] [1585426949.343492603]: Setting RTAB-Map parameter "GridGlobal/FootprintRadius"="0.18"
[ INFO] [1585426949.346692505]: Setting RTAB-Map parameter "GridGlobal/FullUpdate"="true"
[ INFO] [1585426949.351020269]: Setting RTAB-Map parameter "GridGlobal/MaxNodes"="0"
[ INFO] [1585426949.355949786]: Setting RTAB-Map parameter "GridGlobal/MinSize"="1"
[ INFO] [1585426949.360626633]: Setting RTAB-Map parameter "GridGlobal/OccupancyThr"="0.55"
[ INFO] [1585426949.364278243]: Setting RTAB-Map parameter "GridGlobal/ProbClampingMax"="0.971"
[ INFO] [1585426949.367933630]: Setting RTAB-Map parameter "GridGlobal/ProbClampingMin"="0.1192"
[ INFO] [1585426949.372032531]: Setting RTAB-Map parameter "GridGlobal/ProbHit"="0.7"
[ INFO] [1585426949.374898229]: Setting RTAB-Map parameter "GridGlobal/ProbMiss"="0.4"
[ INFO] [1585426949.378927532]: Setting RTAB-Map parameter "GridGlobal/UpdateError"="0.01"
[ERROR] [1585426949.444898849]: Header from subscriber did not have the required elements: md5sum, topic, callerid
[ERROR] [1585426949.445193950]: Header from subscriber did not have the required elements: md5sum, topic, callerid
[ INFO] [1585426949.656163830]: Setting RTAB-Map parameter "Mem/IncrementalMemory"="true"
[ INFO] [1585426949.657284619]: Setting RTAB-Map parameter "Mem/InitWMWithAllNodes"="false"
[ INFO] [1585426950.352882701]:  ...  ZED ready
[ INFO] [1585426950.353022092]: ZED SDK running on GPU #0
[ INFO] [1585426950.367437943]: Camera-IMU Transform: 
 5590BC0AB0
0.999951 0.009705 0.001980 -0.002000
-0.009708 0.999952 0.001471 -0.023000
-0.001966 -0.001490 0.999997 -0.002000
0.000000 0.000000 0.000000 1.000000

[ INFO] [1585426950.367551702]:  * CAMERA MODEL	 -> ZED 2
[ INFO] [1585426950.367668372]:  * Serial Number -> 28477986
[ INFO] [1585426950.367745524]:  * Camera FW Version -> 1523
[ INFO] [1585426950.367824339]:  * Sensors FW Version -> 776
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
[ INFO] [1585426951.311111961]: Advertised on topic /zed/zed_node/rgb/image_rect_color
[ INFO] [1585426951.311245080]: Advertised on topic /zed/zed_node/rgb/camera_info
Error:   Received an invalid TCPROS header.  Each line must have an equals sign.
         at line 103 in /tmp/binarydeb/ros-melodic-cpp-common-0.6.13/src/header.cpp
[ INFO] [1585426951.521295901]: Advertised on topic /zed/zed_node/rgb_raw/image_raw_color
[ INFO] [1585426951.521391260]: Advertised on topic /zed/zed_node/rgb_raw/camera_info
[ INFO] [1585426951.646806540]: Advertised on topic /zed/zed_node/left/image_rect_color
[ INFO] [1585426951.646960107]: Advertised on topic /zed/zed_node/left/camera_info
[ INFO] [1585426951.832532621]: Advertised on topic /zed/zed_node/left_raw/image_raw_color
[ INFO] [1585426951.832833322]: Advertised on topic /zed/zed_node/left_raw/camera_info
[ INFO] [1585426951.963406948]: Advertised on topic /zed/zed_node/right/image_rect_color
[ INFO] [1585426951.963491524]: Advertised on topic /zed/zed_node/right/camera_info
[ INFO] [1585426952.115912637]: Advertised on topic /zed/zed_node/right_raw/image_raw_color
[ INFO] [1585426952.116008092]: Advertised on topic /zed/zed_node/right_raw/camera_info
[ INFO] [1585426952.275555661]: Advertised on topic /zed/zed_node/rgb/image_rect_gray
[ INFO] [1585426952.275772363]: Advertised on topic /zed/zed_node/rgb/camera_info
[ INFO] [1585426952.393683657]: RTAB-Map detection rate = 1.000000 Hz
[ INFO] [1585426952.416485214]: Reading parameters from the ROS server...
[ INFO] [1585426952.435812790]: rtabmap: Deleted database "/home/nvidia/.ros/rtabmap.db" (--delete_db_on_start or -d are set).
[ INFO] [1585426952.435943925]: rtabmap: Using database from "/home/nvidia/.ros/rtabmap.db" (0 MB).
[ INFO] [1585426952.443132267]: Advertised on topic /zed/zed_node/rgb_raw/image_raw_gray
[ INFO] [1585426952.443222570]: Advertised on topic /zed/zed_node/rgb_raw/camera_info
[ INFO] [1585426952.585798953]: rtabmap: Database version = "0.19.3".
[ INFO] [1585426952.684514702]: Advertised on topic /zed/zed_node/left/image_rect_gray
[ INFO] [1585426952.684599629]: Advertised on topic /zed/zed_node/left/camera_info
[ INFO] [1585426952.909330749]: /zed/rtabmap: subscribe_depth = true
[ INFO] [1585426952.910001078]: /zed/rtabmap: subscribe_rgb = true
[ INFO] [1585426952.910179828]: /zed/rtabmap: subscribe_stereo = false
[ INFO] [1585426952.910241587]: /zed/rtabmap: subscribe_rgbd = false (rgbd_cameras=1)
[ INFO] [1585426952.910373778]: /zed/rtabmap: subscribe_odom_info = false
[ INFO] [1585426952.910441297]: /zed/rtabmap: subscribe_user_data = false
[ INFO] [1585426952.910512176]: /zed/rtabmap: subscribe_scan = false
[ INFO] [1585426952.910565488]: /zed/rtabmap: subscribe_scan_cloud = false
[ INFO] [1585426952.910617519]: /zed/rtabmap: queue_size    = 10
[ INFO] [1585426952.910670671]: /zed/rtabmap: approx_sync   = true
[ INFO] [1585426952.910858669]: Setup depth callback
[ INFO] [1585426952.928226234]: Advertised on topic /zed/zed_node/left_raw/image_raw_gray
[ INFO] [1585426952.928352536]: Advertised on topic /zed/zed_node/left_raw/camera_info
[ INFO] [1585426953.025819338]: 
/zed/rtabmap subscribed to (approx sync):
   /zed/zed_node/rgb/image_rect_color,
   /zed/zed_node/depth/depth_registered,
   /zed/zed_node/rgb/camera_info
[ INFO] [1585426953.115764874]: rtabmap 0.19.3 started...
[ INFO] [1585426953.136668850]: Advertised on topic /zed/zed_node/right/image_rect_gray
[ INFO] [1585426953.137064654]: Advertised on topic /zed/zed_node/right/camera_info
[ INFO] [1585426953.279867181]: Advertised on topic /zed/zed_node/right_raw/image_raw_gray
[ INFO] [1585426953.279984588]: Advertised on topic /zed/zed_node/right_raw/camera_info
[ INFO] [1585426953.396436762]: Advertised on topic /zed/zed_node/depth/depth_registered
[ INFO] [1585426953.396526297]: Advertised on topic /zed/zed_node/depth/camera_info
[ INFO] [1585426953.492813080]: Advertised on topic /zed/zed_node/stereo/image_rect_color
[ INFO] [1585426953.580642606]: Advertised on topic /zed/zed_node/stereo_raw/image_raw_color
[ INFO] [1585426953.583849005]: Advertised on topic /zed/zed_node/confidence/confidence_map
[ INFO] [1585426953.585609274]: Advertised on topic /zed/zed_node/disparity/disparity_image
[ INFO] [1585426953.589831599]: Advertised on topic /zed/zed_node/point_cloud/cloud_registered
[ INFO] [1585426953.591704444]: Advertised on topic /zed/zed_node/pose
[ INFO] [1585426953.594285025]: Advertised on topic /zed/zed_node/pose_with_covariance
[ INFO] [1585426953.598159257]: Advertised on topic /zed/zed_node/odom
[ INFO] [1585426953.602385389]: Advertised on topic /zed/zed_node/path_odom
[ INFO] [1585426953.604939475]: Advertised on topic /zed/zed_node/path_map
[ INFO] [1585426953.607441529]: Advertised on topic /zed/zed_node/imu/data @ 400 Hz
[ INFO] [1585426953.611449968]: Advertised on topic /zed/zed_node/imu/data_raw @ 400 Hz
[ INFO] [1585426953.615284680]: Advertised on topic /zed/zed_node/imu/mag @ 50 Hz
[ INFO] [1585426953.620912206]: Advertised on topic /zed/zed_node/temperature/imu @ 400 Hz
[ INFO] [1585426953.624417930]: Advertised on topic /zed/zed_node/atm_press @ 25 Hz
[ INFO] [1585426953.629184537]: Advertised on topic /zed/zed_node/temperature/left @ 25 Hz
[ INFO] [1585426953.632245817]: Advertised on topic /zed/zed_node/temperature/right @ 25 Hz
[ INFO] [1585426953.636580205]: Advertised on topic /zed/zed_node/left_cam_imu_transform [LATCHED]
[ INFO] [1585426953.687689373]: Parameters read = 311
[ INFO] [1585426953.687790300]: Parameters successfully read.
[ERROR] [1585426953.698609709]: Header from subscriber did not have the required elements: md5sum, topic, callerid
[ INFO] [1585426953.711410089]: Static transform Sensor to Base [zed_left_camera_frame -> base_link]
[ INFO] [1585426953.711552487]:  * Translation: {0.000,-0.060,0.000}
[ INFO] [1585426953.711623910]:  * Rotation: {0.000,-0.000,0.000}
[ INFO] [1585426953.711707589]: Static transform Sensor to Camera Center [zed_left_camera_frame -> zed_camera_center]
[ INFO] [1585426953.711766213]:  * Translation: {0.000,-0.060,0.000}
[ INFO] [1585426953.711930115]:  * Rotation: {0.000,-0.000,0.000}
[ INFO] [1585426953.712075554]: Static transform Camera Center to Base [zed_camera_center -> base_link]
[ INFO] [1585426953.712155201]:  * Translation: {0.000,0.000,0.000}
[ INFO] [1585426953.712268448]:  * Rotation: {0.000,-0.000,0.000}
[ INFO] [1585426953.714363114]: *** Starting Positional Tracking ***
[ INFO] [1585426953.714460137]:  * Waiting for valid static transformations...
[ INFO] [1585426953.815046363]: Initial ZED left camera pose (ZED pos. tracking): 
[ INFO] [1585426953.815409880]:  * T: [0,0.06,0]
[ INFO] [1585426953.815590198]:  * Q: [0,0,0,1]
[ INFO] [1585426954.010053983]: Odometry aligned to last tracking pose
[ INFO] [1585426954.441073508]: rtabmap (1): Rate=1.00s, Limit=0.000s, RTAB-Map=0.6086s, Maps update=0.0003s pub=0.0002s (local map=1, WM=1)
[ INFO] [1585426954.745213253]: /zed/rtabmapviz: subscribe_depth = false
[ INFO] [1585426954.745344771]: /zed/rtabmapviz: subscribe_rgb = false
[ INFO] [1585426954.745663296]: /zed/rtabmapviz: subscribe_stereo = false
[ INFO] [1585426954.745731775]: /zed/rtabmapviz: subscribe_rgbd = false (rgbd_cameras=1)
[ INFO] [1585426954.745864350]: /zed/rtabmapviz: subscribe_odom_info = false
[ INFO] [1585426954.746130107]: /zed/rtabmapviz: subscribe_user_data = false
[ INFO] [1585426954.746976050]: /zed/rtabmapviz: subscribe_scan = false
[ INFO] [1585426954.747054354]: /zed/rtabmapviz: subscribe_scan_cloud = false
[ INFO] [1585426954.747093489]: /zed/rtabmapviz: queue_size    = 10
[ INFO] [1585426954.747124785]: /zed/rtabmapviz: approx_sync   = true
[ INFO] [1585426954.747232560]: Setup scan callback
[ INFO] [1585426954.753088115]: 
/zed/rtabmapviz subscribed to:
   /zed/zed_node/odom
[ INFO] [1585426954.753418928]: rtabmapviz started.
[ INFO] [1585426955.354111170]: rtabmap (2): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5025s, Maps update=0.0212s pub=0.0001s (local map=1, WM=1)
[ INFO] [1585426956.413327111]: rtabmap (3): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5165s, Maps update=0.0085s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426957.595936064]: rtabmap (4): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5897s, Maps update=0.0076s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426958.571186488]: rtabmap (5): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5290s, Maps update=0.0086s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426959.635672866]: rtabmap (6): Rate=1.00s, Limit=0.000s, RTAB-Map=0.6084s, Maps update=0.0081s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426960.585699853]: rtabmap (7): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5313s, Maps update=0.0082s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426961.603676134]: rtabmap (8): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5038s, Maps update=0.0081s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426962.664763151]: rtabmap (9): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5370s, Maps update=0.0093s pub=0.0001s (local map=1, WM=1)
[ INFO] [1585426963.634575142]: rtabmap (10): Rate=1.00s, Limit=0.000s, RTAB-Map=0.4597s, Maps update=0.0087s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426964.733464318]: rtabmap (11): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5637s, Maps update=0.0090s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426965.794250885]: rtabmap (12): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5838s, Maps update=0.0102s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426966.800608608]: rtabmap (13): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5732s, Maps update=0.0087s pub=0.0000s (local map=1, WM=1)
[ INFO] [1585426967.749048979]: rtabmap (14): Rate=1.00s, Limit=0.000s, RTAB-Map=0.4959s, Maps update=0.0078s pub=0.0001s (local map=1, WM=1)
[ INFO] [1585426968.744456558]: rtabmap (15): Rate=1.00s, Limit=0.000s, RTAB-Map=0.5037s, Maps update=0.0091s pub=0.0000s (local map=1, WM=1)
^C[zed/rviz-6] killing on exit
[zed/rtabmap-5] killing on exit
[zed/rtabmapviz-4] killing on exit
[ INFO] [1585426969.156333481]: rtabmapviz: ctrl-c catched! Exiting Qt app...
[zed/zed_node-3] killing on exit
[zed/zed_state_publisher-2] killing on exit
[ INFO] [1585426969.374495547]: rtabmapviz stopping spinner...
Saving parameters to /home/nvidia/.ros/rtabmap.cfg
rtabmap: Saving database/long-term memory... (located at /home/nvidia/.ros/rtabmap.db)
rtabmap: Saving database/long-term memory...done! (located at /home/nvidia/.ros/rtabmap.db, 31 MB)
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done

```

These are my launch files:

**zed_rtabmap.launch**
(I changed this line:
`        <include file="$(find zed_wrapper)/launch/include/zed_camera.launch.xml">`
)

```
<?xml version="1.0"?>
<!--
Copyright (c) 2018, STEREOLABS.

All rights reserved.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
    <arg name="zed_namespace"		    	default="zed" />

    <arg name="svo_file"                        default="" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->

    <arg name="zed_node_name"                   default="zed_node" />
    <arg name="camera_model"                    default="zed2" /> <!-- 'zed' or 'zedm' or 'zed2' -->
    <arg name="publish_urdf"                    default="true" />

    <group ns="$(arg zed_namespace)">
        <!-- ZED Wrapper Node-->
        <include file="$(find zed_wrapper)/launch/include/zed_camera.launch.xml">
            <arg name="svo_file"            value="$(arg svo_file)" />
            <arg name="node_name"           value="$(arg zed_node_name)" />
            <arg name="camera_model"        value="$(arg camera_model)" />
            <arg name="publish_urdf"        value="$(arg publish_urdf)" />
        </include>
    
  
        <!-- RTAB-Map -->
        <arg name="custom_rtabmap_launch_file" default="$(find zed_rtabmap_example)/launch/include/sl_rtabmap.launch.xml"/>
        <include file="$(arg custom_rtabmap_launch_file)">
            <arg name="odom_topic"              default="zed_node/odom" />
            <arg name="rgb_topic"               default="zed_node/rgb/image_rect_color" />
            <arg name="depth_topic"             default="zed_node/depth/depth_registered" />
            <arg name="camera_info_topic"       default="zed_node/rgb/camera_info" />
            <arg name="depth_camera_info_topic" default="zed_node/depth/camera_info" />
        </include>   

        <!-- Rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find zed_rtabmap_example)/rviz/zed-rtabmap.rviz" output="screen" /> 
    </group>
</launch>
```

**zed_camera.launch.xml**
```
<?xml version="1.0"?>
<!--
Copyright (c) 2020, STEREOLABS.

All rights reserved.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
    <!-- Camera Model and Name -->
    <arg name="camera_name"           default="zed" /> <!-- The name you want -->
    <arg name="camera_model"          default="zed2" /> <!-- 'zed' or 'zedm' or 'zed2' -->
    <arg name="node_name"             default="zed_node" />

    <!-- Load SVO file -->
    <arg name="svo_file"              default="" /><!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
    <!-- Remote stream -->
    <arg name="stream"                default="" /> <!-- <arg name="stream" default="<ip_address>:<port>"> -->

    <!-- Base frame -->
    <arg name="base_frame"            default="base_link" />

    <!-- Publish ZED urdf -->
    <arg name="publish_urdf"          default="true" />

    <arg name="camera_id"             default="0" />
    <arg name="gpu_id"                default="-1" />

    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_x"             default="0.0" />
    <arg name="cam_pos_y"             default="0.0" />
    <arg name="cam_pos_z"             default="0.0" />

    <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0" />
    <arg name="cam_pitch"             default="0.0" />
    <arg name="cam_yaw"               default="0.0" />

    <!-- ROS URDF description of the ZED -->
    <group if="$(arg publish_urdf)">
        <param name="$(arg camera_name)_description"
               command="$(find xacro)/xacro '$(find zed_wrapper)/urdf/zed_descr.urdf.xacro'
                        camera_name:=$(arg camera_name)
                        camera_model:=$(arg camera_model)
                        base_frame:=$(arg base_frame)
                        cam_pos_x:=$(arg cam_pos_x)
                        cam_pos_y:=$(arg cam_pos_y)
                        cam_pos_z:=$(arg cam_pos_z)
                        cam_roll:=$(arg cam_roll)
                        cam_pitch:=$(arg cam_pitch)
                        cam_yaw:=$(arg cam_yaw)"
        />

        <node name="$(arg camera_name)_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" required="true">
            <remap from="robot_description" to="$(arg camera_name)_description" />
        </node>
    </group>

    <node name="$(arg node_name)" pkg="zed_wrapper" type="zed_wrapper_node" output="screen" required="true"><!-- launch-prefix="valgrind" -->
        <rosparam file="$(find zed_wrapper)/params/common.yaml" command="load" />
        <rosparam file="$(find zed_wrapper)/params/$(arg camera_model).yaml" command="load" />

        <!-- Camera name -->
        <param name="general/camera_name"               value="$(arg camera_name)" />

        <!-- Base frame -->
        <param name="general/base_frame"                value="$(arg base_frame)" />

        <!-- SVO file path -->
        <param name="svo_file"                          value="$(arg svo_file)" />

        <!-- Remote stream -->
        <param name="stream"                            value="$(arg stream)" />

        <!-- Camera ID -->
        <param name="general/zed_id"                    value="$(arg camera_id)" />

        <!-- GPU ID -->
        <param name="general/gpu_id"                    value="$(arg gpu_id)" />
    </node>
</launch>

```

**sl_rtabmap.launch.xml**

```
<!-- RTAB-Map launcher -->

<launch>
    <!-- Localization-only mode -->
    <arg name="localization"            default="false"/>

    <arg     if="$(arg localization)" name="args" default=""/>
    <arg unless="$(arg localization)" name="args" default="--delete_db_on_start"/>

    <!-- RGB-D related topics -->
    <arg name="rgb_topic"               default="zed2/zed_node/rgb/image_rect_color" />
    <arg name="depth_topic"             default="zed2/zed_node/depth/depth_registered" />
    <arg name="camera_info_topic"       default="zed2/zed_node/rgb/camera_info" />
    <arg name="depth_camera_info_topic" default="zed2/zed_node/depth/camera_info" />

    <arg name="odom_topic"              default="zed2/zed_node/odom" />    <!-- Odometry topic name -->
    <!--<arg name="odom_topic"              default="odom" />-->

    <!-- Visual Odometry -->
    <!--
    <node name="rtabmap_stereo_odometry" pkg="rtabmap_ros" type="stereo_odometry" output="screen" args="" launch-prefix="">    
        <remap from="/left/image_rect"      to="zed/zed_node/left/image_rect_color"/>
        <remap from="/right/image_rect"     to="zed/zed_node/right/image_rect_color"/>
        <remap from="/left/camera_info"     to="zed/zed_node/left/camera_info"/>
        <remap from="/right/camera_info"    to="zed/zed_node/right/camera_info"/>
    </node>
    -->

    <!-- RTABmapviz -->
    <node name="rtabmapviz" pkg="rtabmap_ros" type="rtabmapviz" output="screen" args="" launch-prefix="">    
        <remap from="rgb/image"         to="$(arg rgb_topic)"/>
        <remap from="depth/image"       to="$(arg depth_topic)"/>
        <remap from="rgb/camera_info"   to="$(arg camera_info_topic)"/>

        <remap from="grid_map"          to="map" />
        <remap from="odom"              to="$(arg odom_topic)"/>
    </node> 


    <!-- RTABmap -->
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="$(arg args)" launch-prefix="">
        <rosparam command="load" file="$(find zed_rtabmap_example)/params/rtabmap.yaml" />
    
        <remap from="rgb/image"         to="$(arg rgb_topic)"/>
        <remap from="depth/image"       to="$(arg depth_topic)"/>
        <remap from="rgb/camera_info"   to="$(arg camera_info_topic)"/>

        <remap from="grid_map"          to="map" />
        <remap from="odom"              to="$(arg odom_topic)"/>

        <!-- localization mode -->
        <param     if="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="false"/>
        <param unless="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="true"/>
        <param name="Mem/InitWMWithAllNodes" type="string" value="$(arg localization)"/>
    </node>

</launch>

```

**rtabmap version**

```nvidia@nvidia-desktop:~/.ros$ rosversion rtabmap
0.19.3
nvidia@nvidia-desktop:~/.ros$ rosversion rtabmap_ros
0.19.3
```

rviz and rtabviz images:
![Screenshot from 2020-03-28 15-28-30](https://user-images.githubusercontent.com/25854555/77833149-25244600-7109-11ea-9734-1811c76a2786.png)
![Screenshot from 2020-03-28 15-29-22](https://user-images.githubusercontent.com/25854555/77833158-2fdedb00-7109-11ea-9ee2-bc99142807a0.png)

Thank you.

--- a/examples/zed_rtabmap_example/launch/zed_rtabmap.launch
+++ b/examples/zed_rtabmap_example/launch/zed_rtabmap.launch
@@ -17,20 +17,26 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 -->
 <launch>
-    <arg name="zed_namespace"		    	default="zed" />
+    <arg name="svo_file"             default="" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
+    <arg name="stream"               default="" /> <!-- <arg name="stream" default="<ip_address>:<port>"> -->
 
-    <arg name="svo_file"                        default="" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
+    <arg name="zed_node_name"        default="zed_node" />
+    <arg name="camera_model"         default="zed2" />
+    <arg name="publish_urdf"         default="true" />
 
-    <arg name="zed_node_name"                   default="zed_node" />
-    <arg name="camera_model"                    default="zed2" /> <!-- 'zed' or 'zedm' or 'zed2' -->
-    <arg name="publish_urdf"                    default="true" />
+    <arg name="camera_name"          default="zed" />
 
-    <group ns="$(arg zed_namespace)">
+    <arg name="base_frame"           default="base_link" />
+
+    <group ns="$(arg camera_name)">
         <!-- ZED Wrapper Node-->
-        <include file="$(find zed_wrapper)/launch/zed_camera.launch">
+        <include file="$(find zed_wrapper)/launch/include/zed_camera.launch.xml">
+            <arg name="camera_name"         value="$(arg camera_name)" />
             <arg name="svo_file"            value="$(arg svo_file)" />
+            <arg name="stream"              value="$(arg stream)" />
             <arg name="node_name"           value="$(arg zed_node_name)" />
             <arg name="camera_model"        value="$(arg camera_model)" />
+            <arg name="base_frame"          value="$(arg base_frame)" />
             <arg name="publish_urdf"        value="$(arg publish_urdf)" />
         </include>
     
@@ -38,11 +44,11 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
         <!-- RTAB-Map -->
         <arg name="custom_rtabmap_launch_file" default="$(find zed_rtabmap_example)/launch/include/sl_rtabmap.launch.xml"/>
         <include file="$(arg custom_rtabmap_launch_file)">
-            <arg name="odom_topic"              default="zed_node/odom" />
-            <arg name="rgb_topic"               default="zed_node/rgb/image_rect_color" />
-            <arg name="depth_topic"             default="zed_node/depth/depth_registered" />
-            <arg name="camera_info_topic"       default="zed_node/rgb/camera_info" />
-            <arg name="depth_camera_info_topic" default="zed_node/depth/camera_info" />
+            <arg name="odom_topic"              default="$(arg zed_node_name)/odom" />
+            <arg name="rgb_topic"               default="$(arg zed_node_name)/rgb/image_rect_color" />
+            <arg name="depth_topic"             default="$(arg zed_node_name)/depth/depth_registered" />
+            <arg name="camera_info_topic"       default="$(arg zed_node_name)/rgb/camera_info" />
+            <arg name="depth_camera_info_topic" default="$(arg zed_node_name)/depth/camera_info" />
         </include>   
 
         <!-- Rviz -->

--- a/examples/zed_rtabmap_example/rviz/zed-rtabmap.rviz
+++ b/examples/zed_rtabmap_example/rviz/zed-rtabmap.rviz
@@ -87,6 +87,10 @@ Visualization Manager:
           Alpha: 1
           Show Axes: false
           Show Trail: false
+        zed_baro_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
         zed_camera_center:
           Alpha: 1
           Show Axes: false
@@ -100,11 +104,23 @@ Visualization Manager:
           Alpha: 1
           Show Axes: false
           Show Trail: false
+        zed_mag_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
         zed_right_camera_frame:
           Alpha: 1
           Show Axes: false
           Show Trail: false
         zed_right_camera_optical_frame:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        zed_temp_left_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        zed_temp_right_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false

